### DISCUSSION TITLE
cannot use termux-api command in proot

### DISCUSSION THREAD

**Discussion by Teerapath-P101** [termux/termux-app#3358] [General]
(my English may not be very good)

I tried to use termux-api command in proot and this has happened.

![Screenshot_2023-04-24-07-30-37-87_84d3000e3f4017145260f7618db1d683](https://user-images.githubusercontent.com/85441805/233875894-437ce1ab-859f-4002-9ea5-05081f25469d.jpg)
CANNOT LINK EXECUTABLE "/system/bin/app_process": library "liboplusplugin.so" not found: needed by /system/lib64/libandroid_runtime.so in namespace (default)

I've tried following a similar discussions but it still doesn't work for me I don't know what I should do.

About my termux info
Termux Variables:
TERMUX_API_VERSION=0.50.1
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP_PID=15786
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_VERSION=0.118.0
Packages CPU architecture:
aarch64
Subscribed repositories:
 sources.list
deb https://packages.termux.org/apt/termux-main/ stable main
Updatable packages:
All packages up to date
termux-tools version:
1.38.1
Android version:
13
Kernel build information:
Linux localhost 5.10.110-android12-9-00011-g8efa13274425-ab9212784 #1 SMP PREEMPT Mon Oct 24 22:45:30 UTC 2022 aarch64 Android
Device manufacturer:
realme
Device model:
RMX3561
LD Variables:
LD_LIBRARY_PATH=
LD_PRELOAD=

**Comment by MatejMagat305** [comment]
so, in proot (chroot) processes do not see outside, it is isolated enviroment which can not see termux script from api ... 
but if I understand you have ubuntu or debian and you have instaled deb from termux, that is good idea, but proot is set to new path, so if you like experiments, you can try set  LD_PRELOAD= to android path of shared library and then try this again

**Comment by sylirre** [comment]
You have to bind all necessary system directories inside PRoot in order to run Termux:API. You also need to set certain environment variables, no this is not LD_PRELOAD or LD_LIBRARY_PATH. Particularly this is what done by `proot-distro` to let `am`, `termux-api` and other things work.

**Reply by Teerapath-P101** [reply]
> You have to bind all necessary system directories inside PRoot in order to run Termux:API. You also need to set certain environment variables

Thank you for your reply. But I'm not sure what commands and environment variables I should use And should I set it to .bashrc?. can you tell me about that 
sorry, I'm newbie.😅
